pkg/reconciler/git/clone.go

```diff
@@ -156,3 +159,29 @@ func mapSecretKey(URL string) (string, error) {
 
 	return output, nil
 }
+
+func (r *Cloner) ResolveRevision(rev string) (*gitp.Hash, error) {
+	auth, err := r.buildAuth()
+	if err != nil {
+		return nil, err
+	}
+	// in case this is called before the repo is cloned, we must have a clone for the resolver to work.
+	repo := r.repoClient.Repo()
+	if repo == nil {
+		repo, err = git.Clone(memory.NewStorage(), nil, &git.CloneOptions{
```

Comment:
Means, in the current implementation we are cloning always twice, because we are first retreiving the revision and afterwards triggering the real clone, right?

Reply:
Let's have tomorrow a quick sync, as we agreed today to make the version optional. I think we can easily introduce this new behaviour into your PR which will also fix the side-effect we currently have with resolving first the revision and cloning the sources later again.